src/fetch/payouts.py

```diff
@@ -468,9 +468,13 @@ def construct_payouts(
     batch_rewards_df = batch_rewards_df.drop(
         ["partner_list", "partner_fee_eth"], axis=1
     )
+
+    assert batch_rewards_df["solver"].is_unique, "solver not unique in batch rewards"
```

Comment:
Although i guess this test will always pass, it would be nice to have the error msg report the set of non-unique entries. Not a deal-breaker for this PR though

Reply:
We should add that when we rewrite the code.